pluginbase/src/statusmanager.cpp

```diff
@@ -0,0 +1,94 @@
+#include "statusmanager.h"
+#include <QApplication>
+#include <QLoggingCategory>
+#include <utility>
+
+using namespace scopy;
+
+Q_LOGGING_CATEGORY(CAT_STATUSMANAGER, "StatusManager")
+
+StatusManager *StatusManager::pinstance_{nullptr};
+
+StatusManager::StatusManager(QObject *parent)
+	: m_timer(new QTimer(this))
+	, m_itemQueue(new QList<StatusMessage *>)
+	, m_enabled(false)
+{
+	connect(this, &StatusManager::messageAdded, this, &StatusManager::processStatusMessage);
+	connect(m_timer, &QTimer::timeout, this, [this]() {
+		m_timer->stop();
+		Q_EMIT clearDisplay();
+		processStatusMessage();
+	});
+}
+
+StatusManager::~StatusManager() { delete m_itemQueue; }
+
+StatusManager *StatusManager::GetInstance()
+{
+	if(pinstance_ == nullptr) {
+		pinstance_ = new StatusManager(QApplication::instance()); // singleton has the app as parent
+	}
+	return pinstance_;
+}
+
+void StatusManager::addTemporaryMessage(const QString &message, int ms)
```

Comment:
To simply call StatusBarManager::pushMessage without getting the instance first, you can declare a static pushMessage and a private _pushMessage() member

The static method gets the instance and calls the member. This allows for cleaner singleton interfacing.
Smth similar is implemented here:

https://github.com/analogdevicesinc/scopy/pull/1500/files#diff-00efcbdc18feab83b204658845a8715cbf6d323ffd89c7216d3a866e2ff73ca2R31-R34